### PR TITLE
chore(warehouse): added append mode to Deltalake destination

### PR DESCRIPTION
**Pull request by achettyiitr**
# Description

Add support for deltalake append mode.

## Notion Ticket

https://www.notion.so/rudderstacks/Enable-append-mode-for-deltalake-destination-f0e00b62d9584f0888f2e9a723998641

## Security

- [x] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: warehouse/deltalake/deltalake.go [hunk 1]
```diff
@@ -603,7 +602,7 @@ func (dl *HandleT) loadUserTables() (errorMap map[string]error) {
 		strings.Join(firstValProps, ","),
 		warehouseutils.UsersTable,
 		identifyStagingTable,
-		quotedUserColNames,
+		columnNames(userColNames),
```
Comment (satishrudderstack) [inline warehouse/deltalake/deltalake.go:605]: @achettyiitr Is this related to current change ?

File: warehouse/deltalake/deltalakeQueryBuilder.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+		columnsWithValues(columnKeys),
+		columnNames(columnKeys),
+		stagingColumnNames(columnKeys),
+	)
+	return
+}
+
+func (*LTAppend) SqlStatement(namespace string, tableName string, stagingTableName string, columnKeys []string) (sqlStatement string) {
+	stagingTableSqlStatement := stagingSqlStatement(namespace, tableName, stagingTableName, columnKeys)
+	sqlStatement = fmt.Sprintf(`INSERT INTO %[1]s.%[2]s (%[4]s) SELECT %[4]s FROM ( %[5]s ) AS r;`,
```
Comment (satishrudderstack) [inline warehouse/deltalake/deltalakeQueryBuilder.go:65]: @achettyiitr Minor: Do we need aliasing here 'AS r' ?

File: warehouse/deltalake/deltalake.go [hunk 3]
```diff
@@ -548,6 +541,13 @@ func (dl *HandleT) loadTable(tableName string, tableSchemaInUpload warehouseutil
 	return
 }
 
+func loadTableHandler() LoadTable {
+	if loadTableStrategy == "APPEND" {
+		return &LTAppend{}
+	}
+	return &LTMerge{}
```
Comment (abhimanyubabbar) [inline warehouse/deltalake/deltalake.go:548]: If we are ever going to have two different strategy like append and merge and functionality is only limited to generating the sql statement different, should we represent them with simple conditional statement like
```
   if strategy == 'APPEND':
       return appendableSQLStatement()
   return mergeableSQLStatement()
```

Wdyt ?
I am saying this as this would prevent us to have unnecessary abstraction. Strategy pattern is mostly valid when you have many strategies and enough capabilities defined within a strategy.

**Review by satishrudderstack — APPROVED** [review]
LGTM

File: warehouse/deltalake/deltalake.go [hunk 4]
```diff
@@ -515,30 +518,19 @@ func (dl *HandleT) loadTable(tableName string, tableSchemaInUpload warehouseutil
...
-	// Getting the primary key for the merge sql statement
-	primaryKey := "id"
-	if column, ok := primaryKeyMap[tableName]; ok {
-		primaryKey = column
+	columnKeys := sortedColumnKeys
+	if loadTableStrategy == "APPEND" {
+		columnKeys = warehouseutils.SortColumnKeysFromColumnMap(tableSchemaAfterUpload)
+	}
+	if loadTableStrategy == "APPEND" {
+		sqlStatement = appendableLTSQLStatement(dl.Namespace, tableName, stagingTableName, columnKeys)
```
Comment (abhimanyubabbar) [inline warehouse/deltalake/deltalake.go:526]: Should we club them together  in a single if block ?

**Review by abhimanyubabbar — APPROVED** [review]
LGTM 🎈 🥳 🎉
